multimodal/src/autogluon/multimodal/learners/semantic_segmentation.py

```diff
@@ -2,21 +2,128 @@
 import os
 from typing import Dict, Iterable, List, Optional, Union
 
+import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
 import torch
+import torchmetrics
 from omegaconf import OmegaConf
 from PIL import Image
-
-from ..constants import LABEL, LOGITS
-from ..optimization.utils import get_metric, get_norm_layer_param_names, get_trainable_params_efficient_finetune
-from ..utils import extract_from_output, setup_save_path
+from scipy.special import softmax
+
+from autogluon.core.metrics import Scorer
+
+from ..constants import LABEL, LOGITS, MASK_SEMANTIC_INFER, SEMANTIC_SEGMENTATION
+from ..models import get_model_postprocess_fn
+from ..optimization.lit_semantic_segmentation import SemanticSegmentationLitModule
+from ..optimization.utils import (
+    get_loss_func,
+    get_metric,
+    get_norm_layer_param_names,
+    get_trainable_params_efficient_finetune,
+)
+from ..utils import (
+    create_fusion_data_processors,
+    create_fusion_model,
+    extract_from_output,
+    get_dir_ckpt_paths,
+    get_load_ckpt_paths,
+    setup_save_path,
+)
 from .base import BaseLearner
 
 logger = logging.getLogger(__name__)
 
 
 class SemanticSegmentationLearner(BaseLearner):
+    def __init__(
+        self,
+        label: Optional[str] = None,
+        problem_type: Optional[str] = SEMANTIC_SEGMENTATION,
+        presets: Optional[str] = None,
+        eval_metric: Optional[Union[str, Scorer]] = None,
+        hyperparameters: Optional[dict] = None,
+        path: Optional[str] = None,
+        verbosity: Optional[int] = 2,
+        num_classes: Optional[int] = None,  # TODO: can we infer this from data?
+        classes: Optional[list] = None,
+        warn_if_exist: Optional[bool] = True,
+        enable_progress_bar: Optional[bool] = None,
+        pretrained: Optional[bool] = True,
+        validation_metric: Optional[str] = None,
+        sample_data_path: Optional[str] = None,
+        **kwargs,
+    ):
+        super().__init__(
+            label=label,
+            problem_type=problem_type,
+            presets=presets,
+            eval_metric=eval_metric,
+            hyperparameters=hyperparameters,
+            path=path,
+            verbosity=verbosity,
+            num_classes=num_classes,
+            classes=classes,
+            warn_if_exist=warn_if_exist,
+            enable_progress_bar=enable_progress_bar,
+            pretrained=pretrained,
+            validation_metric=validation_metric,
+            sample_data_path=sample_data_path,
+        )
+        self._output_shape = num_classes
+        self._sample_data_path = sample_data_path
+
+        if self._output_shape == None:
+            self._output_shape = 1  # binary semantic segmentation by default
+
+        if self._sample_data_path is not None:
+            self._output_shape = self.get_semantic_segmentation_class_num(self._sample_data_path)
```

Comment:
What if users provide both num_classes and sample data path? Do we need to verify they are consistent?

Reply:
What if users neither provide num_classes and sample data path? In this case, we need to infer from training data.